test/unit/full_text_search/similar_search_issue_test.rb

```diff
@@ -0,0 +1,117 @@
+      fts_engine_groonga_open_source =
+        perform_enqueued_jobs(only: FullTextSearch::UpdateIssueContentJob) do
+          Issue.generate!(
+            project: @project,
+            subject: "ぐるんが",
+            description: "高速に検索できます。 オープンソースです。")
+        end
+      fts_engine_pgroonga_open_source =
+        perform_enqueued_jobs(only: FullTextSearch::UpdateIssueContentJob) do
+          Issue.generate!(
+            project: @project,
+            subject: "ぴーじーるんが",
+            description: "PostgreSQLに組み込んで高速に検索できます。 オープンソースです。")
+        end
+
+      similar_issues =
+        fts_engine_groonga_open_source.similar_issues(
+          project_ids: [@project.id])
+      assert_equal([fts_engine_pgroonga_open_source],
+                   similar_issues)
+
+      perform_enqueued_jobs(only: FullTextSearch::UpdateIssueContentJob) do
+        fts_engine_groonga_open_source.update!(description: nil)
+      end
+      similar_issues =
+        fts_engine_groonga_open_source.similar_issues(
+          project_ids: [@project.id])
+      assert_equal([], similar_issues)
+    end
+
+    def test_same_structure_with_journal
+      fts_engine_groonga_open_source =
+        perform_enqueued_jobs(only: FullTextSearch::UpdateIssueContentJob) do
+          Issue.generate!(project: @project, subject: "ぐるんが")
+               .journals.create!(notes: "高速に検索できます。 オープンソースです。")
+        end
+      fts_engine_pgroonga_open_source =
+        perform_enqueued_jobs(only: FullTextSearch::UpdateIssueContentJob) do
+          Issue.generate!(project: @project, subject: "ぴーじーるんが")
+               .journals.create!(
+                 notes: "PostgreSQLに組み込んで高速に検索できます。 オープンソースです。")
+        end
+      target_issue = fts_engine_groonga_open_source.issue
+      similar_issues = target_issue.similar_issues(project_ids: [@project.id])
+      assert_equal([fts_engine_pgroonga_open_source.issue],
+                   similar_issues)
+
+      perform_enqueued_jobs(only: FullTextSearch::UpdateIssueContentJob) do
+        target_issue.journals.first.destroy
+      end
+      similar_issues =
+        target_issue.reload.similar_issues(project_ids: [@project.id])
+      assert_equal([], similar_issues)
+    end
+
+    def test_same_structure_with_attachment
+      set_tmp_attachments_directory
+      fts_engine_groonga =
+        Issue.generate!(project: @project, subject: "ぐるんが")
+      fts_engine_groonga.save_attachments(
+        [
+          {
+            "file" => mock_file_with_options(:original_filename => "test.txt"),
+            "description" => "高速に検索 オープンソース! 最新情報"
+          }
+        ]
+      )
+      perform_enqueued_jobs(only: FullTextSearch::UpdateIssueContentJob) do
+        fts_engine_groonga.save!
+      end
+      fts_engine_pgroonga =
+        Issue.generate!(project: @project, subject: "ぴーじーるんが")
+      fts_engine_pgroonga.save_attachments(
+        [
+          {
+            "file" => mock_file_with_options(:original_filename => "test.txt"),
```

Comment:
Can we use different `original_filename`?

Reply:
fix: [e097ef2](https://github.com/clear-code/redmine_full_text_search/pull/167/commits/e097ef2dbf7632d1d046a932994b945fed7e45ff)